modules/platform/nuxeo-drive-server/nuxeo-drive-mongodb/src/main/java/org/nuxeo/drive/mongodb/MongoDBAuditChangeFinder.java

```diff
@@ -0,0 +1,187 @@
+        long clusteringDelay = getClusteringDelay(repositoryNames);
+        MongoDBAuditBackend auditService = (MongoDBAuditBackend) Framework.getService(AuditReader.class);
+        QueryBuilder queryBuilder = new AuditQueryBuilder().predicate(
+                Predicates.in(LOG_REPOSITORY_ID, repositoryNames));
+        if (clusteringDelay > -1) {
+            // Double the delay in case of overlapping, see https://jira.nuxeo.com/browse/NXP-14826
+            long lastClusteringInvalidationDate = System.currentTimeMillis() - 2 * clusteringDelay;
+            queryBuilder.and(Predicates.lt(LOG_LOG_DATE, lastClusteringInvalidationDate));
+        }
+        queryBuilder.order(OrderByExprs.desc(LOG_ID));
+        queryBuilder.limit(1);
+        List<LogEntry> entries = auditService.queryLogs(queryBuilder);
+
+        if (entries.isEmpty()) {
+            if (clusteringDelay > -1) {
+                // Check for existing entries without the clustering invalidation date filter to not return -1 in this
+                // case and make sure the lower bound of the next call to NuxeoDriveManager#getChangeSummary will be >=
+                // 0
+                if (auditService.getAuditCollection().estimatedDocumentCount() == 0) {
+                    log.debug("Found no audit log entries matching the criterias but some exist, returning 0");
+                    return 0;
+                }
+            }
+            log.debug("Found no audit log entries, returning -1");
+            return -1;
+        }
+        return entries.get(0).getId();
+    }
+
+    @Override
+    protected List<LogEntry> queryAuditEntries(CoreSession session, SynchronizationRoots activeRoots,
+            Set<String> collectionSyncRootMemberIds, long lowerBound, long upperBound, int limit) {
+        MongoDBAuditBackend auditService = (MongoDBAuditBackend) Framework.getService(AuditReader.class);
+        MongoCollection<Document> auditCollection = auditService.getAuditCollection();
+
+        Bson filter = and(eq("repositoryId", session.getRepositoryName()), or(
+                and(or(and(eq("category", "eventDocumentCategory"),
+                        in("eventId", "documentCreated", "documentModified", "documentMoved", "documentCreatedByCopy",
+                                "documentRestored", "addedToCollection", "documentProxyPublished", "documentLocked",
+                                "documentUnlocked", "documentUntrashed")),
+                        and(eq("category", "eventLifeCycleCategory"), eq("eventId", "lifecycle_transition_event"),
+                                ne("docLifeCycle", "deleted"))),
+                        or(regex("docPath", getRootsRegex(activeRoots.getPaths())),
+                                in("docUUID", collectionSyncRootMemberIds))),
+                and(eq("category", "NuxeoDrive"), ne("eventId", "rootUnregistered"))), gt("_id", lowerBound),
+                lte("_id", upperBound));
```

Comment:
This is hard to read, you can't understand what's ANDde or ORed with what. You should use intermediate variables with meaningful names to construct the global filter step by step.